src/mbgl/map/transform.hpp

```diff
@@ -132,22 +132,87 @@ class Transform : private util::noncopyable {
     FreeCameraOptions getFreeCameraOptions() const;
     void setFreeCameraOptions(const FreeCameraOptions& options);
 
+    struct Animation {
+        TimePoint start;
+        Duration duration;
+        AnimationOptions options;
```

Comment:
An `Animation` has a `duration`, but an `Animation` also has an `AnimationsOptions` object with an optional `duration`. I think this needs some more refactoring...